client/state/form/actions.js

```diff
@@ -15,67 +16,47 @@ export const setFormProperty = ( field, value ) => {
 	};
 };
 
-export const goToStep = ( stepIndex ) => {
-	return {
-		type: GO_TO_STEP,
-		step: stepIndex,
-	};
-};
-
-export const nextStep = () => ( dispatch, getState, { callbackURL, nonce, submitMethod, formSchema, formLayout } ) => {
-	const submitForm = ( callback ) => {
-		const setIsSaving = ( value ) => {
-			dispatch( setFormProperty( 'isSaving', value ) );
-			if ( false === value ) {
-				callback();
-			}
-		};
-		const setSuccess = ( value ) => dispatch( setFormProperty( 'success', value ) );
-		const setError = ( value ) => dispatch( setFormProperty( 'errors', value ) );
-
-		dispatch( setFormProperty( 'pristine', true ) );
-		saveForm( setIsSaving, setSuccess, setError, callbackURL, nonce, submitMethod, getState().form.values );
-	};
-
-	const tryAutoAdvance = ( isManualAction ) => {
-		const currentStep = getState().form.currentStep;
-		const currentStepLayout = formLayout[ currentStep ] || {};
+export const backFromSuggestion = () => ( {
+	type: BACK_FROM_SUGGESTION,
+} );
 
-		if ( currentStep >= formLayout.length - 1 ) {
-			return;
-		}
+export const goToStep = ( stepIndex ) => ( dispatch, getState, { formLayout } ) => {
+	const stepLayout = formLayout[ stepIndex ] || {};
 
-		if ( getStepFormErrors( getState(), formSchema, formLayout ).length ) {
-			return;
-		}
+	// Clear the "bypass suggestion" flag if the user is coming *back* to this step
+	if ( getState().form.currentStep > stepIndex && stepLayout.bypass_suggestion_flag ) {
```

Comment:
`bypass_suggestion_flag` contains the *name* of the field that's used to indicate that the suggestion will be ignored. For example, for the Origin Address step:
`bypass_suggestion_flag: 'orig_bypass_suggestion'`

So if the field `orig_bypass_suggestion` is True (or "true-ish"), then the suggestion will be ignored. That's why the `bypass_suggestion_flag` has `flag` in its name, because it's *not* storing a boolean value, but giving the name of the actual *flag* that contains the boolean value.

I don't know how to name it to make it less confusing. `skip_showing_suggestions` makes it look like a boolean value  :(